pywhy_graphs/algorithms/semi_directed_paths.py

```diff
@@ -0,0 +1,188 @@
+import networkx as nx
+
+from ..config import EdgeType
+from ..typing import Node
+
+__all__ = [
+    "is_semi_directed_path",
+    "all_semi_directed_paths",
+]
+
+
+def _empty_generator():
+    yield from ()
+
+
+def is_semi_directed_path(G, nodes):
+    """Returns True if and only if `nodes` form a semi-directed path in `G`.
+
+    A *semi-directed path* in a graph is a nonempty sequence of nodes in which
+    no node appears more than once in the sequence, each adjacent
+    pair of nodes in the sequence is adjacent in the graph and where each
+    pair of adjacent nodes does not contain a directed endpoint in the direction
+    towards the start of the sequence.
+
+    That is ``(a -> b o-> c <-> d -> e)`` is not a semi-directed path from ``a`` to ``e``
+    because ``d *-> c`` is a directed endpoint in the direction towards ``a``.
+
+    Parameters
+    ----------
+    G : graph
+        A mixed-edge graph.
+    nodes : list
+        A list of one or more nodes in the graph `G`.
+
+    Returns
+    -------
+    bool
+        Whether the given list of nodes represents a simple path in `G`.
```

Comment:
Shouldn't it be a semi-directed path? Or are both the terms synonyms?

Reply:
Thanks good catch!